src/main/java/org/jfrog/hudson/pipeline/common/types/buildInfo/Env.java

```diff
@@ -73,11 +69,49 @@ public Env collectBuildParameters(Run<?, ?> build, TaskListener listener) {
      */
     public IncludeExcludePatterns filter() {
         IncludeExcludePatterns pattern = filter.getPatternFilter();
-        this.envVars.entrySet().removeIf(key -> PatternMatcher.pathConflicts(key.getKey(), pattern));
-        this.sysVars.entrySet().removeIf(key -> PatternMatcher.pathConflicts(key.getKey(), pattern));
+        filterVars(this.envVars, pattern);
+        filterVars(this.sysVars, pattern);
         return pattern;
     }
 
+    /**
+     * Filter environment/system variables if their key matches the include/exclude pattern filters, or if their value contains a suspected secret.
+     *
+     * @param vars    map of environment/system variables
+     * @param pattern include/exclude patterns
+     */
+    private void filterVars(Map<String, String> vars, IncludeExcludePatterns pattern) {
+        vars.entrySet().removeIf(key -> PatternMatcher.pathConflicts(key.getKey(), pattern));
+        vars.entrySet().removeIf(key -> containsSuspectedSecrets(key.getValue()));
+    }
+
+    /**
+     * Checks whether the value of a variable contains at least one suspected secret.
+     * A suspected secret can be an API-Key, access token or reference token.
+     *
+     * @param value - string to search in
+     * @return whether a secret is suspected
+     */
+    private boolean containsSuspectedSecrets(String value) {
```

Comment:
Let's export `AKCp8`, `cmVmdGtuOjAxOj`, and `eyJ2ZXIiOiIyIiwidHlwIjoiSldUIiwiYWxnIjoiUlMyNTYiLCJraWQiOiJ` to static variables. Same for the minimal lengths. These static variables describe by their names what prefixes they are.
One option is to save all in String and int static variables. Another option is using a struct `SecretPrefix` containing the secret prefix and minimal length.